### ISSUE TITLE
PMIX linking error when building --enable-static --disable-shared

### ISSUE THREAD

**Issue by jsquyres** [open-mpi/ompi#1281]
When building master with `--disable-shared --enable-static`, it seems that a PMIX library can't be found when compiling MPI applications:

```sh
$ mpicc -g    hello_c.c   -o hello_c
/usr/bin/ld: cannot find -lpmix
collect2: error: ld returned 1 exit status
$ mpicc --showme -g hello_c.c -o hello_c
gcc -g hello_c.c -o hello_c -I/home/jsquyres/bogus/include -pthread -Wl,-rpath -Wl,/home/jsquyres/bogus/lib -Wl,--enable-new-dtags -L/home/jsquyres/bogus/lib -lmpi -lopen-rte -lopen-pal -lm -lnuma -lpciaccess -libverbs -lrdmacm -ldl -lpmix -lmunge -lutil -lrt
$ ls $bogus/lib/*pmix*
zsh: no matches found: /home/jsquyres/bogus/lib/*pmix*
```

I note that when I build the same way (statically) on the v2.x branch, I get a pmix library, and therefore building MPI executables succeeds:

```
$ ls $bogus/lib/*pmix*
/home/jsquyres/bogus/lib/libpmix.a  /home/jsquyres/bogus/lib/libpmix.la*
```

**Comment by rhc54** [comment]
I believe you just need to remove that -lpmix from the args - does it then work?

**Comment by jsquyres** [comment]
Yes -- if I manually remove `-lpmix` from the mpicc wrapper data text file, the compilation succeeds and the application runs correctly.

Is this all that is necessary?

```diff
diff --git a/opal/mca/pmix/pmix112/configure.m4 b/opal/mca/pmix/pmix112/configur
index b03c54d..59b2950 100644 
--- a/opal/mca/pmix/pmix112/configure.m4 
+++ b/opal/mca/pmix/pmix112/configure.m4  
@@ -61,9 +61,6 @@ AC_DEFUN([MCA_opal_pmix_pmix112_CONFIG],[ 
                   AC_SUBST([opal_pmix_pmix112_LIBS]) 
                   AC_SUBST([opal_pmix_pmix112_CPPFLAGS])])  
  
-           # Finally, add a flag to support static builds
-           pmix_pmix112_WRAPPER_EXTRA_LIBS=-lpmix
- 
            CFLAGS=$opal_pmix_pmix112_save_CFLAGS 
            CPPFLAGS=$opal_pmix_pmix112_save_CPPFLAGS 
            LDFLAGS=$opal_pmix_pmix112_save_LDFLAGS 
diff --git a/opal/mca/pmix/pmix120/configure.m4 b/opal/mca/pmix/pmix120/configur
index c5f94af..4053632 100644 
--- a/opal/mca/pmix/pmix120/configure.m4 
+++ b/opal/mca/pmix/pmix120/configure.m4 
@@ -61,9 +61,6 @@ AC_DEFUN([MCA_opal_pmix_pmix120_CONFIG],[
                   AC_SUBST([opal_pmix_pmix120_LIBS]) 
                   AC_SUBST([opal_pmix_pmix120_CPPFLAGS])]) 
  
-           # Finally, add a flag to support static builds 
-           pmix_pmix120_WRAPPER_EXTRA_LIBS=-lpmix
-
            CFLAGS=$opal_pmix_pmix120_save_CFLAGS 
            CPPFLAGS=$opal_pmix_pmix120_save_CPPFLAGS 
            LDFLAGS=$opal_pmix_pmix120_save_LDFLAGS
```

**Comment by rhc54** [comment]
Should be, yes

**Comment by jsquyres** [comment]
Fixed via e4bdad09c1bf7f11dada5ae6ac32e052b553ce4b.
